docs/modules/ROOT/pages/cop_api_v1_changelog.adoc

```diff
@@ -0,0 +1,258 @@
+    end
+  end
+end
+----
+
+=== If your class supports autocorrection
+
+Your class must `extend Autocorrector`.
+
+The `corrector` is now yielded from `add_offense`. Move the code of your method `autocorrect` in that block and do not wrap your correction in a lambda. `Corrector` are more powerful and can now be `merge`d.
+
+==== Example:
+
+[source,ruby]
+----
+# Before
+class MySillyCorrectingCop < Cop
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node, location: :selector, message: 'Be positive')
+    end
+  end
+
+  def autocorrect(node)
+    lambda do |corrector|
+      corrector.replace(node.loc.selector, '+')
+    end
+  end
+end
+
+# After
+class MySillyCorrectingCop < Base
+  extend Autocorrector
+
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node.loc.selector, message: 'Be positive') do |corrector|
+        corrector.replace(node.loc.selector, '+')
+      end
+    end
+  end
+end
+----
+
+=== If your class uses instance variables
+
+Do not use RuboCop's internal instance variables. If you used `@processed_source`, use `processed_source`. If you have a need to access an instance variable, open an issue with your use case.
+
+Cop instances will be called on different `processed_source` if the class supports it. In all cases it is recommended to to invalidate your cache data during the `on_new_investigation` callback:
+
+[source,ruby]
+----
+# Before
+class MyCachingCop < Cop
+  def on_send(node)
+    if my_cached_data[node]
+      #...
+    end
+  end
+
+  def my_cached_data
+    @data ||= processed_source.comments.map { # ... }
+  end
+end
+
+# After
+class MyCachingCop < Base
+  # ... same as above, plus...
+  def on_new_investigation
+    @data = nil
+    super  # Be nice and call super for callback
+  end
+end
+----
+
+=== Other API changes
+
+If your cop uses `investigate`, `investigate_post_walk`, `join_force?`, or internal classes like `Corrector`, `Commissioner`, `Team`, these have changed. See the Detailed API Changes.
+
+=== Upgrading specs
+
+It is highly recommended you use `expect_offense` / `expect_correction` / `expect_no_offense` in your specs, e.g.:
+
+[source,ruby]
+----
+require 'rubocop/rspec/support'
+
+RSpec.describe RuboCop::Cop::Custom::MySillyCorrectingCop, :config do
+  # No need for `let(:cop)`
+  it 'is positive' do
+    expect_offense(<<~RUBY)
+      42 + 2 - 2
+             ^ Be positive
+    RUBY
+
+    expect_correction(<<~RUBY)
+      42 + 2 + 2
+    RUBY
+  end
+
+  it 'does not register an offense for calls to `despair`' do
+    expect_no_offenses(<<~RUBY)
+      "don't".despair
+    RUBY
+  end
+end
+----
+
+In the unlikely case where you use the class `RuboCop::Cop::Corrector` directly, it has changed a bit but you can ease your transition with `RuboCop::Cop::Legacy::Corrector` that is meant to be somewhat backwards compatible. You will need to `require 'rubocop/cop/legacy/corrector'`.
+
+= Detailed API Changes
```

Comment:
You can't have two top-level headings, so you either have to increase the nesting on split the document in two.

Reply:
I indented one level, but anyone can feel free to split it if they feel it's better.